### ISSUE TITLE
build erros when i run - cargo install ncspot

### ISSUE THREAD

**Issue by thewh1teagle** [hrkfdn/ncspot#190]
debian buster 10
cargo ver. 1.43.0 

first you need to add in readme that we need to install libdbus-glib-1-dev  (in debian requirements section)

after i solved that, i get this error: 

```
$ cargo install ncspot

error[E0308]: mismatched types
  --> /home/qlorg/.cargo/registry/src/github.com-1ecc6299db9ec823/librespot-protocol-0.1.1/build.rs:45:39
   |
45 |         let p = parse_and_typecheck(&["proto"], &[src]).expect("protoc");
   |                                       ^^^^^^^ expected struct `std::path::Path`, found `str`
   |
   = note: expected reference `&std::path::Path`
              found reference `&'static str`

error[E0308]: mismatched types
  --> /home/qlorg/.cargo/registry/src/github.com-1ecc6299db9ec823/librespot-protocol-0.1.1/build.rs:45:51
   |
45 |         let p = parse_and_typecheck(&["proto"], &[src]).expect("protoc");
   |                                                   ^^^ expected struct `std::path::Path`, found struct `std::string::String`
   |
   = note: expected reference `&std::path::Path`
              found reference `&std::string::String`

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0308`.
error: could not compile `librespot-protocol`.

To learn more, run the command again with --verbose.
warning: build failed, waiting for other jobs to finish...
    Building [===========================================>           ] 321/394: darling_core
error: failed to compile `ncspot v0.1.3`, intermediate artifacts can be found at `/tmp/cargo-install7F3VkQ`

Caused by:
  build 

failed

```

**Comment by martinbrennan** [comment]
Hi 
I am seeing the same problem.
Is this to do with the version of the Rust compiler? I have version 1.43.0
Martin

**Comment by hrkfdn** [comment]
It's a bug in an upstream library that was fixed, but not released: https://github.com/hrkfdn/ncspot/issues/179

Try installing with `cargo install ncspot --locked`

**Comment by thewh1teagle** [comment]
> It's a bug in an upstream library that was fixed, but not released: #179
> 
> Try installing with `cargo install ncspot --locked`

That's fixed the problem. i successfuly compiled it.


not i have another bug, when i try to login i get this error


```
thread '<unnamed>' panicked at 'Authentication failed with reason: PremiumAccountRequired', /home/qlorg/.cargo/registry/src/github.com-1ecc6299db9ec823/librespot-core-0.1.1/src/connection/mod.rs:121:21
note: run with `RUST_BACKTRACE=1` environment variable to display
a backtrace
thread '<unnamed>' panicked at 'Authentication failed with reason: BadCredentials', /home/qlorg/.cargo/registry/src/github.com-1ecc6299db9ec823/librespot-core-0.1.1/src/connection/mod.rs:121:21
Didn't obtain any credentials
```

**Comment by martinbrennan** [comment]
Hi Henrik
Thank you for the quick answer and I can confirm that fixed it for me as well
Martin
